### PR TITLE
chore(gatsby-core-utils): ci to TS

### PR DESCRIPTION
**Pull request by danielkov**
Collection ticket for these tasks: #22041
☂ ticket: #21995

### CODE REVIEW

File: packages/gatsby-core-utils/package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "@babel/cli": "^7.7.5",
     "@babel/core": "^7.7.5",
     "babel-preset-gatsby-package": "^0.2.17",
-    "cross-env": "^5.2.1"
+    "cross-env": "^5.2.1",
+    "@types/ci-info": "2.0.0"
```
Comment (MichaelDeBoey) [inline packages/gatsby-core-utils/package.json]: @danielkov You shouldn't manually add dependencies.
They normally appear in alphabetical order

File: packages/gatsby-core-utils/src/__mocks__/ci-info.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
-const ciInfo = jest.genMockFromModule(`ci-info`)
+const ciInfo: typeof import("ci-info") = jest.genMockFromModule(`ci-info`)
```
Comment (blainekasten) [inline packages/gatsby-core-utils/src/__mocks__/ci-info.ts]: is this valid? typically `import()` are async. So this seems off. /shrug

Comment (danielkov) [reply packages/gatsby-core-utils/src/__mocks__/ci-info.ts]: I think you're referring to Webpack's `import()` function that does async import. In TypeScript you can pull in the export signature this way.

File: packages/gatsby-core-utils/package.json (hunk 1 above)
Comment (danielkov) [reply packages/gatsby-core-utils/package.json]: updated

**Pull request merged**